clients/client/src/main/java/org/projectnessie/client/http/HttpClientBuilder.java

```diff
@@ -170,14 +172,29 @@ public HttpClientBuilder withConnectionTimeout(int connectionTimeoutMillis) {
     return this;
   }
 
-  /**
-   * Builds a new {@link NessieHttpClient}.
-   *
-   * @return A new {@link NessieHttpClient}.
-   */
   @Override
-  public NessieClient build() {
-    return new NessieHttpClient(
-        uri, authentication, tracing, readTimeoutMillis, connectionTimeoutMillis);
+  public <API extends NessieApi> API build(NessieApiVersion apiVersion, Class<API> apiContract) {
+    NessieHttpClient client =
+        new NessieHttpClient(
+            uri, authentication, tracing, readTimeoutMillis, connectionTimeoutMillis);
+    API api;
+    switch (apiVersion) {
+      case V_0_9:
+        api = (API) client;
+        break;
+      case V_1:
+        api = (API) new HttpApiV1(client);
```

Comment:
Sorry for the late nit comment. How about `apiContract.cast(new ...)` also on line 183.

Reply:
Also, isn't the API version implied by the `apiContract` class? Why do we need a separate version parameter?